processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -157,6 +163,45 @@ public void remove()
     );
   }
 
+  public static ValueMatcher getTimeValueMatcher(
+      final LongColumnSelector longSelector,
+      Comparable value
+  )
+  {
+    try {
+      if (value == null) {
+        return new BooleanValueMatcher(false);
+      }
+      final long longValue = Long.parseLong(value.toString());
+      return new ValueMatcher()
+      {
+        @Override
+        public boolean matches()
+        {
+          return longSelector.get() == longValue;
+        }
+      };
+    }
+    catch (NumberFormatException nfe) {
+      throw new ParseException("Cannot parse value [%s] as long: ", value);
```

Comment:
If we call this `getLongValueMatcher` so it can be used with other long dims, then this should probably return a `new BooleanValueMatcher(false)` rather than throw an exception. (so a user can write a query that works where some segments are strings and some are longs, and we really do want to select a particular string from the string columns and nothing from the long columns)